app/models/iam_user_identity.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Subclasses the UserIdentity class in order to override the redis namespace and thereby partition
+# user sessions for va.gov from openid client applications.
+#
+class IAMUserIdentity < ::UserIdentity
+  redis_store REDIS_CONFIG[:iam_user_identity][:namespace]
+  redis_ttl REDIS_CONFIG[:iam_user_identity][:each_ttl]
+  redis_key :uuid
+
+  attribute :expiration_timestamp, Integer
+  attribute :iam_icn, String
+  attribute :iam_edipi, String
+  attribute :iam_sec_id, String
+
+  # Builds an identity instance from the profile returned in the IAM introspect response
+  #
+  # @param iam_profile [Hash] the profile of the user, as they are known to the IAM SSOe service.
+  # @return [IAMUserIdentity] an instance of this class
+  #
+  def self.build_from_iam_profile(iam_profile)
+    loa_level = iam_profile[:fediamassur_level].to_i
+
+    identity = new(
+      email: iam_profile[:email],
+      expiration_timestamp: iam_profile[:exp],
```

Comment:
Can we confirm that the format of `exp` is compatible with the call to `redis_namespace.expireat`?

Reply:
Redis namespace gem aliases all the Redis methods and expireat takes a unix timestamp 
https://github.com/redis/redis-rb/blob/master/lib/redis.rb#L430-L439